merlin/systems/dag/ops/tensorflow.py

```diff
@@ -145,35 +144,37 @@ def _export_model(self, model, name, output_path, version=1):
             name=name, backend="tensorflow", platform="tensorflow_savedmodel"
         )
 
-        inputs, outputs = model.inputs, model.outputs
+        # inputs, outputs = model.inputs, [model.outputs]
 
-        if not inputs or not outputs:
-            signatures = getattr(model, "signatures", {}) or {}
-            default_signature = signatures.get("serving_default")
-            if not default_signature:
-                # roundtrip saved model to disk to generate signature if it doesn't exist
+        signatures = getattr(model, "signatures", {}) or {}
+        default_signature = signatures.get("serving_default")
+        if not default_signature:
+            # roundtrip saved model to disk to generate signature if it doesn't exist
 
-                reloaded = tf.keras.models.load_model(tf_model_path)
-                default_signature = reloaded.signatures["serving_default"]
+            reloaded = tf.keras.models.load_model(tf_model_path)
+            default_signature = reloaded.signatures["serving_default"]
+
+        inputs = list(default_signature.structured_input_signature[1].values())
+        outputs = list(default_signature.structured_outputs.values())
 
-            inputs = list(default_signature.structured_input_signature[1].values())
-            outputs = list(default_signature.structured_outputs.values())
+        input_col_names = [*default_signature.structured_input_signature[1].keys()]
+        output_col_names = [*default_signature.structured_outputs.keys()]
 
         config.parameters["TF_GRAPH_TAG"].string_value = "serve"
         config.parameters["TF_SIGNATURE_DEF"].string_value = "serving_default"
 
-        for col in inputs:
+        for col, col_name in zip(inputs, input_col_names):
             config.input.append(
                 model_config.ModelInput(
-                    name=f"{col.name}", data_type=_convert_dtype(col.dtype), dims=[-1, col.shape[1]]
+                    name=col_name, data_type=_convert_dtype(col.dtype), dims=[-1, col.shape[1]]
                 )
             )
 
-        for col in outputs:
+        for col, col_name in zip(outputs, output_col_names):
```

Comment:
maybe the same thing as for inputs? 
```suggestion
        for col, col_name in default_signature.structured_outputs.items():
```